### PR TITLE
#170947384 Implement requests table

### PR DESCRIPTION
**Pull request by blaise82**
## **Why**
The requests table is handy in displaying a users requests.

A logged in user should see the travel requests table along with the details of the requests that they have made.

## **Acceptance Criteria**
**Scenario:** 
```
When a user is on the requests page they should see the requests table.```

Sync with other colleagues implementing other aspects of the page to agree on the details to be seen on the table
```

#### Description of Task to be completed? 
- set up requests redux state
- create table 
- handle pagination
- fetch requests from the database

#### How should this be manually tested?
- open your terminal
- clone this repo by `git clone https://github.com/Stackup-Rwanda/octopus-bn-frontend`
- run `cd octopus-bn-frontend` to move to the project folder
- run `git checkout ` to move to this branch 
- run `npm install` to install all required dependencies for this feature
- run 'npm run dev' to start the development server 

> will open the project on an 8080 port so to access the project you will need to open your browser and load `http://localhost:8080` {port}

- you will be presented with the login page use the credentials below to log in
```
email: octopusbn@gmail.com
password: password
```
- load `http://localhost:8080/requests` or click on `My REQUESTS` option on the drawer to load  `http://localhost:8080/requests

#### Screenshots (if appropriate)
Screen Shot 2020-05-09 at 12 50 36

[starts #170947384]

### CODE REVIEW

**Review by hakoemmy — COMMENTED** [review]
@blaise82 You did an interesting job, Would you please raise your tests coverage.

File: src/redux/reducers/requestsReducer.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+				error: action.payload,
+				open: true
+			};
+		// case CLOSE_MESSAGE:
+		// 	return {
+		// 		loading: 'none',
+		// 		requests: [],
+		// 		error: '',
+		// 		open: false
+		// 	};
```
Comment (hakoemmy) [inline src/redux/reducers/requestsReducer.js]:
```suggestion
	
```
Get a rid of the comments if you they don't add any value

File: src/styles/dashboard.requests.module.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+line-height: 14px;
+/* identical to box height */
+
+letter-spacing: 0.01em;
+
+color: #A6ACBE;
+}
+.button{
+    margin-top: 15px;
+}
```
Comment (hakoemmy) [inline src/styles/dashboard.requests.module.css]:
```suggestion
}

```
EOF

**Review by hakoemmy — COMMENTED** [review]
@blaise82 Address the minor comments left

**Review by hakoemmy — APPROVED** [review]
@blaise82 You did a great job. Keep it up

**Review by Moise1 — CHANGES_REQUESTED** [review]
@blaise82 , please provide a proper message when a logged in user is trying to view their requests yet they have none.

**Review by Moise1 — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

File: src/components/RequestsTable.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	progress: {
+		position: 'absolute',
+		marginRight: 'auto',
+		marginLeft: 'auto',
+		left: 0,
+		right: 0
+	}
+}));
+
+function TablePaginationActions(props) {
```
Comment (AJAkimana) [inline src/components/RequestsTable.js]: Use arrow function

File: src/components/RequestsTable.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+}
+
+TablePaginationActions.propTypes = {
+	count: PropTypes.number.isRequired,
+	onChangePage: PropTypes.func.isRequired,
+	page: PropTypes.number.isRequired,
+	rowsPerPage: PropTypes.number.isRequired
+};
+
+export default function RequestsTable() {
```
Comment (AJAkimana) [inline src/components/RequestsTable.js]: Use an arrow function

File: src/components/ResponsiveDrawer.js [hunk 5]
```diff
@@ -106,9 +109,9 @@ const useStyles = makeStyles((theme) => ({
 		backgroundColor: '#4ba7f2'
 	}
 }));
-
 function ResponsiveDrawer(props) {
```
Comment (AJAkimana) [inline src/components/ResponsiveDrawer.js]: Use arrow function

**Review by AJAkimana — COMMENTED** [review]
As the whole app is using arrow functions, do the same in this PR in order to keep code consistency

**Pull request merged**